templates/core/core-secret.yaml

```diff
@@ -1,18 +1,26 @@
+{{- $secret := include "harbor.core" . }}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ template "harbor.core" . }}
+  name: {{ $secret }}
   labels:
 {{ include "harbor.labels" . | indent 4 }}
 type: Opaque
 data:
-  secretKey: {{ .Values.secretKey | b64enc | quote }}
+{{- if .Release.IsInstall }}
+  secretKey: {{ .Values.core.secretKey | default (randAlphaNum 16) | b64enc | quote }}
   secret: {{ .Values.core.secret | default (randAlphaNum 16) | b64enc | quote }}
+  HARBOR_ADMIN_PASSWORD: {{ .Values.harborAdminPassword | default (randAlphaNum 16) | b64enc | quote }}
+  CSRF_KEY: {{ .Values.core.xsrfKey | default (randAlphaNum 32) | b64enc | quote }}
+{{- else }}
+  secretKey: {{ (lookup "v1" "Secret" .Release.Namespace $secret).data.secretKey }}
```

Comment:
For some keys can passwords that _can_ be changed by the admin after the initial deployment by updating values.yaml (secretKey, secret, CSRF_KEY), I think it makes sense to use the values.yaml value even during an upgrade. This definitely doesn't apply to the harbor admin password.